### ISSUE TITLE
feat: remove the ipv6 check on ILB

### ISSUE THREAD

**Issue by feiskyer** [kubernetes-sigs/cloud-provider-azure#697]
**Is your feature request related to a problem? Please describe.**

ipv6 on ILB would report an error because of the following check:

https://github.com/kubernetes-sigs/cloud-provider-azure/blob/f8cb009de8b8479baa03343308ed88d434a98131/pkg/provider/azure_loadbalancer.go#L1785-L1790

The check should be removed because ipv6 has been supported now on ILB.

**Describe the solution you'd like in detail**

Validate and remove the ipv6 check on ILB.

**Describe alternatives you've considered**

**Additional context**

The changes should also be cherry-picked to k/k.

**Comment by tomkerkhove** [comment]
We are blocked on this so would be nice to have this.

> The changes should also be cherry-picked to k/k.

What do you mean by this @feiskyer?

**Comment by feiskyer** [comment]
@tomkerkhove it means the fixes would also be submitted to kubernetes repo.
